### ISSUE TITLE
Not compatible with PHP 7.2.2

### ISSUE THREAD

**Issue by hwrng** [Sonerezh/sonerezh#334]
After PHP update (nginx 1.13.9 with php-fpm):
```
PHP 7.2.2-1 (cli) (built: Feb  1 2018 15:19:04) ( NTS )
```
Unable to start application any more:

```
[error] 18087#18087: *19 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Cannot use 'Object' as class name as it is reserved in /www/xxx.xxx/lib/Cake/Core/Object.php on line 30

PHP message: PHP Fatal error:  Uncaught Error: Class 'Controller' not found in /www/xxx.xxx/lib/Cake/Error/ExceptionRenderer.php:174
Stack trace:

#0 /www/xxx.xxx/lib/Cake/Error/ExceptionRenderer.php(92): ExceptionRenderer->_getController(Object(InternalErrorException))

#1 /www/xxx.xxx/lib/Cake/Error/ErrorHandler.php(126): ExceptionRenderer->__construct(Object(InternalErrorException))

#2 /www/xxx.xxx/lib/Cake/Error/ErrorHandler.php(284): ErrorHandler::handleException(Object(InternalErrorException))

#3 /www/xxx.xxx/lib/Cake/Error/ErrorHandler.php(213): ErrorHandler::handleFatalError(64, 'Cannot use 'Obj...', '/www/xxx.xxx...', 30)

#4 /www/xxx.xxx/lib/Cake/Core/App.php(970): ErrorHandler::handleError(64, 'Cannot use 'Obj...', '/www/xxx.xxx...', 30, Array)

#5 /www/xxx.xxx/lib/Cake/Core/App.php(943): App::_checkFatalError()

#6 [internal function]: App::shutdown()
```

**Comment by MightyCreak** [comment]
Could you test with https://github.com/gs11/sonerezh?
It's a bit more up to date than this repo.

**Comment by hwrng** [comment]
Yes, gs11 fork is working fine with PHP 7.2.2

**Comment by MightyCreak** [comment]
Will be fixed with #339
